src/Button.js

```diff
@@ -3,100 +3,105 @@ import React from 'react';
-  onClick: PropTypes.func,
-  componentClass: elementType,
-  href: PropTypes.string,
-  /**
-   * Defines HTML button type attribute
-   * @defaultValue 'button'
-   */
-  type: PropTypes.oneOf(['button', 'reset', 'submit', null])
-};
-
-const defaultProps = {
-  active: false,
-  block: false,
-  disabled: false,
-  type: 'button'
-};
-
 class Button extends React.Component {
-  renderAnchor(elementProps, className) {
-    return (
-      <SafeAnchor
-        {...elementProps}
-        className={classNames(className, elementProps.disabled && 'disabled')}
-      />
-    );
-  }
+  static propTypes = {
+    /**
+     * @default 'btn'
+     */
+    bsPrefix: PropTypes.string,
 
-  renderButton({ componentClass, ...elementProps }, className) {
-    const Component = componentClass || 'button';
+    /**
+     * One or more button variant combinations
+     *
+     * buttons may be one of a variety of visual variants such as:
+     *
+     * `'primary', 'secondary', 'success', 'danger', 'warning', 'info, 'dark', 'light', 'link'`
+     *
+     * as well as "outline" versions (prefixed by 'outline-*')
+     *
+     * `'outline-primary', 'outline-secondary', 'outline-success', 'outline-danger', 'outline-warning', 'outline-info, 'outline-dark', 'outline-light'`
+     */
+    variant: PropTypes.string,
 
-    return (
-      <Component
-        {...elementProps}
-        type={elementProps.type}
-        className={className}
-      />
-    );
-  }
+    /**
+     * Specifies a large or small button.
+     *
+     * @type ('sm'|'lg')
+     */
+    size: PropTypes.string,
 
-  render() {
-    const { active, block, className, ...props } = this.props;
-    const [bsProps, elementProps] = splitBsProps(props);
+    /** Spans the full width of the Button parent */
+    block: PropTypes.bool,
+
+    /** Manually set the visual state of the button to `:active` */
+    active: PropTypes.bool,
+
+    /**
+     * Disables the Button, preventing mouse events,
+     * even if the underlying component is an `<a>` element
+     */
+    disabled: PropTypes.bool,
 
-    const classes = {
-      ...getClassSet(bsProps),
+    /** Providing a `href` will render an `<a>` element, _styled_ as a button. */
+    href: PropTypes.string,
+
+    /**
+     * Defines HTML button type attribute.
+     *
+     * @default 'button'
+     */
+    type: PropTypes.oneOf(['button', 'reset', 'submit', null]),
+
+    componentClass: elementType
+  };
+
+  static defaultProps = {
+    variant: 'primary',
+    active: false,
+    disabled: false,
+    type: 'button'
+  };
+
+  render() {
+    const {
+      bsPrefix,
+      variant,
+      size,
       active,
-      [prefix(bsProps, 'block')]: block
-    };
-    const fullClassName = classNames(className, classes);
+      className,
+      block,
+      type,
+      componentClass,
+      ...props
+    } = this.props;
 
-    if (elementProps.href) {
-      return this.renderAnchor(elementProps, fullClassName);
+    delete props.bsRole;
```

Comment:
can we do something like `bsRole: _0` in the spread above, and add a lint rule to not warn on unused vars starting with `_`? i think that's what we're largely doing now, and i think it's good